compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1362,6 +1306,70 @@ private STToken processSlashToken() {
         }
     }
 
+    /**
+     * Process and return documentation string.
+     * <p>
+     * <code>
+     * DocumentationContentString := ( BlankSpace* # [DocumentationContent] )+
+     * <br/>
+     * DocumentationContent := (^ 0xA)* 0xA
+     * <br/>
+     * BlankSpace := Tab | Space
+     * <br/>
+     * Space := 0x20
+     * <br/>
+     * Tab := 0x9
+     * </code>
+     *
+     * @return Documentation string token
+     */
+    private STToken processDocumentationContentString() {
+        int nextChar = peek();
+        while (!reader.isEOF()) {
+            switch (nextChar) {
+                case LexerTerminals.CARRIAGE_RETURN:
+                case LexerTerminals.NEWLINE:
+
+                    // Advance reader for the new line
+                    if (peek() == LexerTerminals.CARRIAGE_RETURN && reader.peek(1) == LexerTerminals.NEWLINE) {
+                        reader.advance();
+                    }
+                    reader.advance();
+
+                    // Look ahead and see if next line also belongs to the documentation.
+                    // i.e. look for a `WS #` match
+                    // If there's a match, advance reader for the next line as well.
+                    // Otherwise terminate documentation content after the new line.
+                    int lookAheadCount = 0;
+                    int lookAheadChar = reader.peek(lookAheadCount);
+                    while (lookAheadChar == LexerTerminals.SPACE || lookAheadChar == LexerTerminals.TAB) {
+                        lookAheadCount++;
+                        lookAheadChar = reader.peek(lookAheadCount);
+                    }
+
+                    if (lookAheadChar != LexerTerminals.HASH) {
+                        // Next line does not belong to documentation, hence break
+                        break;
+                    }
+
+                    reader.advance(lookAheadCount);
+                    nextChar = peek();
+                    continue;
+                default:
+                    reader.advance();
+                    nextChar = peek();
+                    continue;
+            }
+            break;
+        }
+
+        STNode leadingTrivia = STNodeFactory.createNodeList(this.leadingTriviaList);
+        String lexeme = getLexeme();
+        STNode trailingTrivia = STNodeFactory.createNodeList(new ArrayList<>(0)); // No trailing trivia
```

Comment:
New line is already included in the documentation string. 
e.g. 
```
#  line1
function foo() { }
```
documentation string would be `"# line1\n"`
Then in the documentation lexer it will be parsed as the trailing trivia of whatever the last token.